### ISSUE TITLE
RestructuredText files without title does not work anymore

### ISSUE THREAD

**Issue by wandura** [phpDocumentor/phpDocumentor#3005]
https://github.com/phpDocumentor/phpDocumentor/blob/aae2e648cc1c9dad55b8ec7cd8488ec34dbdd05a/src/Guides/RestructuredText/Handlers/ParseFileHandler.php#L116

https://github.com/phpDocumentor/phpDocumentor/blob/54d26db6d7235751b39059b4555bb234b2cc6411/src/Guides/RestructuredText/Handlers/ParseFileHandler.php#L142

This change breaks including rst files without a title like a file with only replace directives.

substitutes.rst
```
.. |product_name| replace:: My Product
.. |company_name| replace:: My Company
```

overview.rst
```
.. include:: substitutes.rst

My Title
========

|product_name| was the first product of |company_name| at the early ...
```

This leads to an error because the TitleNode is null.
And could be fixed with `$document->getTitle() ? $document->getTitle()->getValueString() : '',`.

**Comment by mvriel** [comment]
This issue needs to be reviewed whether it is solved after #3011 . I can imagine the fixes there will also fix this

**Comment by mvriel** [comment]
Not fixed by #3011; to be continued

**Comment by mvriel** [comment]
This issue is now resolved on master; I did find that a substitution containing an underscore is not interpreted. As far as I can see just about all other characters are. I am going to close this issue since the underscore thing may be part of the RST syntax; or an entirely different issue
